### PR TITLE
2.03 add new tag element

### PR DESCRIPTION
**Pull request by allthatilk**
Fixes https://github.com/IATI/IATI-Schemas/issues/342

### CODE REVIEW

**Comment by hayfield** [comment]
PR should have a description.

**Comment by dalepotter** [comment]
~Just had a quick look at this whilst adding test cases for namespace elements and attributes.~ <- Done in 477fed5

~There needs to be test cases for `tag/narrative` too - I will mark as incomplete and add these shortly.~ <- Done in 5127068

**Comment by dalepotter** [comment]
Just had a thought - is there any `should-fail` test that could be applied to `iati-activities/iati-activity/tag/@vocabulary-uri` ?
 The datatype is marked as `xsd:anyURI`.

**Comment by hayfield** [comment]
`xsd:anyURI` is very permissive https://www.w3.org/TR/xmlschema-2/#anyURI

> Because it is impractical for processors to check that a value is a context-appropriate URI reference, this specification follows the lead of [RFC 2396] (as amended by [RFC 2732]) in this matter: such rules and restrictions are not part of type validity and are not checked by ·minimally conforming· processors. Thus in practice the above definition imposes only very modest obligations on ·minimally conforming· processors.

> The ·lexical space· of anyURI is finite-length character sequences which, when the algorithm defined in Section 5.4 of [XML Linking Language] is applied to them, result in strings which are legal URIs according to [RFC 2396], as amended by [RFC 2732].

There are probably a few keyboard-mashes that will violate http://www.ietf.org/rfc/rfc2396.txt and http://www.ietf.org/rfc/rfc2732.txt though.

**Comment by dalepotter** [comment]
We're defining `<xsd:element ref="tag" minOccurs="0" maxOccurs="unbounded"/>`, so I would suggest adding a `should-pass` test case for 2 `tag` elements.

File: iati-activities-schema.xsd [hunk 1]
```diff
@@ -1704,6 +1705,45 @@
     </xsd:complexType>
   </xsd:element>
 
+  <xsd:element name="tag">
+    <xsd:annotation>
+      <xsd:documentation xml:lang="en">
+        Categorisations from established taxonomies that enrich the
```
Comment (hayfield) [inline iati-activities-schema.xsd]: Scout's honour. Rapping.

File: tests/should-fail/iati-activities/iati-activity/tag/err-not-iati-xml-missing-attribute.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+
+<iati-activities version="xx">
+  <iati-activity> <!-- at least 1 -->
+    <iati-identifier></iati-identifier>  <!-- 1 and only 1-->
```
Comment (hayfield) [inline tests/should-fail/iati-activities/iati-activity/tag/err-not-iati-xml-missing-attribute.xml]: Lots of comments like this that should be removed.

Will do a more thorough look once comments are removed (it's hard to do when they're there!)

**Review by hayfield — CHANGES_REQUESTED** [review]

**Comment by dalepotter** [comment]
Marking this as `incomplete` for the time being, pending line wrapping change.

File: iati-activities-schema.xsd (hunk 1 above)
Comment (dalepotter) [reply iati-activities-schema.xsd]: Fixed in fd0e247.

File: tests/should-pass/iati-activities/iati-activity/tag/element-exists-with-required-and-optional-attributes.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+
+<iati-activities version="xx">
+  <iati-activity> <!-- at least 1 -->
```
Comment (dalepotter) [inline tests/should-pass/iati-activities/iati-activity/tag/element-exists-with-required-and-optional-attributes.xml]: These pesky comments still there - grrr...

**Review by dalepotter — APPROVED** [review]

File: tests/should-fail/iati-activities/iati-activity/tag/err-not-iati-xml-missing-attribute.xml (hunk 2 above)
Comment (hayfield) [reply tests/should-fail/iati-activities/iati-activity/tag/err-not-iati-xml-missing-attribute.xml]: 6be59361a38f36b706476abc63a4b30f516afd31

**Review by hayfield — APPROVED** [review]

**Pull request merged**